help/destinations/catalog/email-marketing/sendgrid.md

```diff
@@ -0,0 +1,250 @@
+
+![](../../assets/catalog/email-marketing/sendgrid/11.jpg)
+
+* Now map the *Source* and *Schema* *(created earlier in [prerequisites](#prerequisites-prerequisites))* fields.
+
+![](../../assets/catalog/email-marketing/sendgrid/12.jpg)
+
+> [!IMPORTANT]
+> The *Source* field attribute *"Email"* should be mapped to the *Target* field attribute "external_id" as shown in the subsequent steps.
+![](../../assets/catalog/email-marketing/sendgrid/13.png)
+
+![](../../assets/catalog/email-marketing/sendgrid/14.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/15.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/16.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/17.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/18.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/19.jpg)
+
+* A complete list of the mappings to be setup is shown below.
+
+![](../../assets/catalog/email-marketing/sendgrid/20.png)
+
+![](../../assets/catalog/email-marketing/sendgrid/21.png)
+
+![](../../assets/catalog/email-marketing/sendgrid/22.png)
+
+* After completing the mappings, click *Next* to obtain the below review screen indicating the audience and destination are connected.
+
+![](../../assets/catalog/email-marketing/sendgrid/23.jpg)
+
+* Click *Finish* to complete the setup.
+
+![](../../assets/catalog/email-marketing/sendgrid/24.png)
+
+* Navigate to the list of destinations. Ensure that the destination we created is visible with status enabled. You can also see the flow run status, which is generally updated within an hour.
+
+![](../../assets/catalog/email-marketing/sendgrid/25.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/26.jpg)
+
+* Switch to the *Activation data* tab.
+
+![](../../assets/catalog/email-marketing/sendgrid/27.jpg)
+
+* Monitor the segments to check the flow of data.
+
+![](../../assets/catalog/email-marketing/sendgrid/28.jpg)
+
+* Naviage to the Sendgrid site and check if the new email record from the dataset you created within [prerequisites](#prerequisites-prerequisites) is being populated within the new contact list.
+
+![](../../assets/catalog/email-marketing/sendgrid/29.jpg)
+
+>[!IMPORTANT]
+> * At times, the data might not be visible within Sendgrid immediately due to constraints/limitations at the backend. In such scenarios, please wait for a while before rechecking. 
+> * Alternatively, only a few records might be visible initially, and the remaining data might become visible after an interval.
+
+* Also check a couple of emails to validate if the field mapping is correct.
+
+![](../../assets/catalog/email-marketing/sendgrid/30.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/31.jpg)
+
+![](../../assets/catalog/email-marketing/sendgrid/32.jpg)
+
+## Exported data {#exported-data}
+
+* A sample JSON like the one below is sent by the destination. If you are facing any issues, please ensure the data is compliant.
```

Comment:
One question here: is the data sent in the JSON transparent to customers at any time? I'm wondering if we should keep this information in the user-facing documentation or not.

Reply:
You are correct though useful debug information probably better within the support troubleshooting content,